node/node.go

```diff
@@ -132,19 +132,23 @@ func NewNode(config *cfg.Config,
 	if err != nil {
 		return nil, err
 	}
+
+	genDoc, err := genesisDocProvider()
+	if err != nil {
+		return nil, err
+	}
+
 	state := sm.LoadState(stateDB)
 	if state == nil {
-		genDoc, err := genesisDocProvider()
-		if err != nil {
-			return nil, err
-		}
 		state, err = sm.MakeGenesisState(stateDB, genDoc)
 		if err != nil {
 			return nil, err
 		}
 		state.Save()
+	} else {
+		state.SetChainID(genDoc.ChainID)
```

Comment:
right - we shouldn't have a need for this (but I see this is why the genesis was moved out of the if block. The chainID should be saved somewhere. The ConsensusParams may also change over the life of the chain (not implemented yet), so we don't want to set them back to the genesis params every time we start